skrifa/src/outline/autohint/instance.rs

```diff
@@ -0,0 +1,32 @@
+//! Autohinting state for a font instance.
+
+use super::{super::OutlineGlyphCollection, style::GlyphStyleMap};
+use crate::MetadataProvider;
+use alloc::sync::Arc;
+use raw::TableProvider;
+
+/// Set of derived glyph styles that are used for automatic hinting.
+///
+/// These are invariant per font so can be precomputed and reused for multiple
+/// instances when requesting automatic hinting.
+#[derive(Clone, Debug)]
+pub struct GlyphStyles(Arc<GlyphStyleMap>);
+
+impl GlyphStyles {
+    /// Precomputes the full set of glyph styles for the given outlines.
+    pub fn new(outlines: &OutlineGlyphCollection) -> Self {
+        if let Some(outlines) = outlines.common() {
+            let glyph_count = outlines
+                .font
+                .maxp()
+                .map(|maxp| maxp.num_glyphs() as u32)
+                .unwrap_or_default();
+            Self(Arc::new(GlyphStyleMap::new(
```

Comment:
Do I understand correctly, that an `Arc` type is used here mostly for being able to be able to get the GlyphStyleMap "past the borrow checker" and being able to use it in multiple Skrifa calls in parallel threads? - 

Regarding write-safety: How is that ensured when IIRC info for additional scripts need to be added to the GlyphStyleMap?

What I am trying to say: Could you help me understand why this is an Arc type?

Reply:
Yes, exactly. We want to be able to share the styles among multiple instances (potentially in multiple threads), so `Arc` is The Way to do this in safe Rust. The alternative would be to clone this for each new instance which would require an allocation and copy. 

The style map is computed in full on construction (FreeType does this as well) so it’s never mutated.

The mutable, lazy data structure we talked about is the one that holds unscaled style metrics (alignment zones and widths). That was added in #1077 and uses `RwLock` to handle updates.

Reply:
Thanks, got it. TIL about `RwLock`, how handy.